tests/unit_tests/test_matchmaker_queue_algorithm.py

```diff
@@ -43,6 +56,19 @@ def test_stable_marriage():
     assert (s2, s5) in matches
     assert (s3, s6) in matches
 
+def test_stable_marriage_matches_new_players_with_new_and_old_with_old():
+    new1 = Search([p(1500, 500, name='new1', ladder_games=1)])
+    new2 = Search([p(1400, 500, name='new2', ladder_games=2)])
+    old1 = Search([p(2300, 75, name='old1', ladder_games=100)])
+    old2 = Search([p(2350, 75, name='old2', ladder_games=200)])
+
+    searches = [new1, new2, old1, old2]
+
+    matches = algorithm.stable_marriage(searches)
+
+    assert (new1, new2) in matches
+    assert (old1, old2) in matches
+
```

Comment:
Out of curiosity, what about when 4 players have the same mean but different deviation? 

```
new1 = Search([p(1500, 500, name='new1', ladder_games=0)])
new2 = Search([p(1500, 500, name='new2', ladder_games=0)])
old1 = Search([p(500, 75, name='old1', ladder_games=100)])
old2 = Search([p(500, 75, name='old2', ladder_games=100)])
```
500 would be the adjusted mean for players with 0 games.

I would guess the new players should match with each other and the experienced players with each other.

Reply:
This test should be expanded, you're right.

Reply:
See `test_stable_marriage_matches_new_players_with_new_and_old_with_old_if_same_mean`.